From 378c3845424fafcce3df96a3091d7a44a98e6107 Mon Sep 17 00:00:00 2001
From: pieterg <pieterg@users.sourceforge.net>
Date: Sat, 11 Jan 2014 14:37:21 +0100
Subject: [PATCH] tcp_offload: disable TSQ optimization for GSO

Which turns out to have a negative performance effect on the bcmgenet driver

This effectively reverts d6a4a10411764cf1c3a5dad4f06c5ebe5194488b
---
 net/ipv4/tcp_offload.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/ipv4/tcp_offload.c b/net/ipv4/tcp_offload.c
index 533c58a..75c9cb2 100644
--- a/net/ipv4/tcp_offload.c
+++ b/net/ipv4/tcp_offload.c
@@ -69,7 +69,7 @@ struct sk_buff *tcp_tso_segment(struct sk_buff *skb,
 		goto out;
 	}
 
-	copy_destructor = gso_skb->destructor == tcp_wfree;
+	copy_destructor = false;
 	ooo_okay = gso_skb->ooo_okay;
 	/* All segments but the first should have ooo_okay cleared */
 	skb->ooo_okay = 0;
-- 
1.7.9.5

